crates/ignore/src/lib.rs

```diff
@@ -196,6 +196,27 @@ impl Error {
         }
     }
 
+    /// Returns the `ErrorKind` for programmatic decisions.
+    pub fn io_kind(&self) -> Option<std::io::ErrorKind> {
+        match *self {
+            Error::Partial(ref errs) => {
+                if errs.len() == 1 {
+                    errs[0].io_kind()
+                } else {
+                    None
+                }
+            }
+            Error::WithLineNumber { ref err, .. } => err.io_kind(),
+            Error::WithPath { ref err, .. } => err.io_kind(),
+            Error::WithDepth { ref err, .. } => err.io_kind(),
+            Error::Loop { .. } => None,
+            Error::Io(ref err) => Some(err.kind()),
+            Error::Glob { .. } => None,
+            Error::UnrecognizedFileType(_) => None,
+            Error::InvalidDefinition => None,
+        }
+    }
```

Comment:
I think it would probably be better to just expose a `&std::io::Error`, no? It gives strictly more information and also provides access to a `std::io::ErrorKind`. Might as well provide an `into_io_error` as well.

This is how walkdir does it: https://docs.rs/walkdir/2.3.1/walkdir/struct.Error.html#method.io_error

Reply:
Good to know there is a baseline to be consistent with.  I copied things over from there.

When writing this, I was split on which direction to go.  I saw the `ErrorKind` being useful for programmatic use and `Error` being useful for error propagation but wasn't as clear what the value of a `&Error` would be.  However, there is value in not being prescriptive and letting the user decide what has value.  And then comes along consistency :)